security-force-monitor/sfm-cms
# add upload logic to makefile + change download view and ui

## Overview
For issue:
- #807 

Change include:
- new recipe in Makefile for uploading the zip archive
- added the `awscli` dependency so we can use the aws cli in the docker container
- new dependency for `make recreate_db` so we upload the fresh zip whenever we import data on the server
- changed the `DownloadData` view to serve up a presigned url for the zip archive
- changed the download ui to prompt a download of the zip archive

### Demo
<img width="1433" alt="Screen Shot 2022-07-01 at 10 11 06 AM" src="https://user-images.githubusercontent.com/38969506/176921576-2acc2fdc-e33c-4ab5-9c79-dd5aa987e624.png">


### Notes
See comment offline regarding AWS access.

I created a new S3 bucket in the SFM account. `make data_archive` will overwrite the existing zip archive anytime we run it, no matter local, staging, or production environment. Do we want to have separate buckets per environment? Note that `make download_to_s3` will only run whenever `make recreate_db` runs on the server — that's a command that we never run in local development.

## Testing Instructions
- Pull down this branch
- Run `docker-compose run --rm app make data_archive`. it'll take a few minutes to download all of the data.
- Once the data is downloaded, visit [the download page](http://localhost:8000/en/download/)
- Confirm the download link works
- Confirm the downloaded data is as expected
